### ISSUE TITLE
 init: exit not detected with 'expect stop' and 'respawn' 

### ISSUE THREAD

**Issue by CameronNemo** [CameronNemo/upstart#16]
https://bugs.launchpad.net/upstart/+bug/712351

**Comment by CameronNemo** [comment]
https://bugs.launchpad.net/upstart/+bug/1182943

**Comment by socketpair** [comment]
Yes, I have solution for case when, say, daemon with “expect stop” dies before sending sigstop to itself. Seems this fix helps for cases when daemon dies inthe middleof doubleforking or so.

I will make pull request shortly, I use this patchinproduction for a while in Centos 6. Sorry, holidays (Victory Day in Russia) and I’m not at PC.

**Comment by socketpair** [comment]
Please ping me if I forget

**Comment by socketpair** [comment]
This works for upstart  0.6.5 (Centos 6).

**Comment by socketpair** [comment]
@CameronNemo Please review

**Comment by CameronNemo** [comment]
I merged something similar. It is here: https://gitlab.freedesktop.org/CameronNemo/upstart/commit/ac745396d4bb8c3efb66ac2d749415a2d9874467

**Comment by CameronNemo** [comment]
Are there any other issues you have been hitting / think are important? These are my high priority ones: https://gitlab.freedesktop.org/CameronNemo/upstart/issues?milestone_title=1.13.3

**Comment by CameronNemo** [comment]
I would still like to add a test in `init/tests/test_job_process.c` that fails without the fix but succeeds with it.
